SimulationAgent/DeviceProperties/SetDeviceTag.cs

```diff
@@ -1,52 +1,50 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 using System;
-using System.Threading.Tasks;
 using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
 using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
-using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
 
-namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceConnection
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceProperties
 {
     /// <summary>
     /// Add twin information to the new device twin
     /// </summary>
-    public class DeviceTwinTag : IDeviceConnectionLogic
+    public class SetDeviceTag : IDevicePropertiesLogic
     {
         private readonly IDevices devices;
         private readonly ILogger log;
         private string deviceId;
-        private IDeviceConnectionActor context;
+        private IDevicePropertiesActor context;
 
-        public DeviceTwinTag(IDevices devices, ILogger logger)
+        public SetDeviceTag(IDevices devices, ILogger logger)
         {
             this.log = logger;
             this.devices = devices;
         }
 
-        public void Setup(IDeviceConnectionActor context, string deviceId, DeviceModel deviceModel)
+        public void Setup(IDevicePropertiesActor context, string deviceId)
         {
             this.context = context;
             this.deviceId = deviceId;
         }
 
-        public async Task RunAsync()
+        public void Run()
         {
             this.log.Debug("Adding tag to device twin...", () => new { this.deviceId });
             try
             {
                 var now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
-                await this.devices.AddTagAsync(this.deviceId);
+                this.devices.AddTagAsync(this.deviceId);
```

Comment:
Previously we were waiting for the async operation to finish now we are not. Any reason?

Reply:
Because the DevicePropertiesActor is on its own thread, the state machine can execute synchronously without holding up the other actor operations. The Run() operations in the other actors follow this pattern and so I also updated this class to follow that pattern here:

![image](https://user-images.githubusercontent.com/3317135/38570465-caa70164-3ca2-11e8-852c-ccdb6a36c251.png)


You can see another example of it in the DeviceTelemetryActor here:
https://github.com/Azure/device-simulation-dotnet/blob/7449fde2a82f74a6ec298de231c848885aaf5f05/SimulationAgent/DeviceTelemetry/SendTelemetry.cs#L39

Reply:
I had a comment in the previous review, suggesting to use `ContinueWith`, like in `SendTelemetry.cs`, otherwise the app could call HandleEvent (line 42) before catching the exception (I'm not sure the exceptions would be caught either)